examples/example-add-vectors.cpp

```diff
@@ -0,0 +1,164 @@
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <iostream>
+
+#include "memoryManager.hpp"
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/defines.hpp"
+
+const int CUDA_BLOCK_SIZE = 256;
+
+void checkSolution(int *C, int in_N);
+
+/*
+  Example 1: Adding Two Vectors
+
+  ----[Details]---------------------
+  Adds two vectors of length N
+
+  -----[RAJA Concepts]---------------
+  1. Introduces the forall loop and basic RAJA policies
+
+  RAJA::forall<exec_policy>(iter_space I, [=] (index_type i)) {
+
+         //body
+
+  });
+
+  [=] By-copy capture
+  [&] By-reference capture (for non-unified memory targets)
+  exec_policy - Specifies how the traversal occurs
+  iter_space  - Iteration space for RAJA loop (any random access container is expected)
+  index_type  - Index for RAJA loops
+ */
+int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
+{
+
+  printf("Example 1: Adding Two Vectors \n \n");
+
+  const int N = 1000;
+  int *A = memoryManager::allocate<int>(N);
+  int *B = memoryManager::allocate<int>(N);
+  int *C = memoryManager::allocate<int>(N);
+
+  for (int i = 0; i < N; ++i) {
+    A[i] = -i;
+    B[i] =  i;
+  }
+
+  printf("Standard C++ Loop \n");
+  for (int i = 0; i < N; ++i) {
+    C[i] = A[i] + B[i];
+  }
+  checkSolution(C, N);
+
+
+  printf("RAJA: Sequential Policy \n");
+  /*
+    RAJA::seq_exec -  Executes the loop sequentially
+
+    RAJA::RangeSegment(start,stop) - Generates a contiguous sequence of numbers
+    by the [start, stop) interval specified
+  */
+  RAJA::forall<RAJA::seq_exec>(
+    RAJA::RangeSegment(0, N), [=](RAJA::Index_type i) {
+      C[i] = A[i] + B[i]; 
+    });
+  checkSolution(C, N);
+  
+  
+#if defined(RAJA_ENABLE_OPENMP)
+  printf("RAJA: OpenMP Policy \n");
+  /*
+    RAJA::omp_parallel_for_exec - executes the forall loop using the
+    #pragma omp parallel for directive
+  */
+  RAJA::forall<RAJA::omp_parallel_for_exec>(
+    RAJA::RangeSegment(0, N), [=](RAJA::Index_type i) {
+      C[i] = A[i] + B[i];
+    });
+  checkSolution(C, N);
+#endif
+  
+  
+#if defined(RAJA_ENABLE_CUDA)
+  printf("RAJA: CUDA Policy \n");
+  /*
+    RAJA::cuda_exec<CUDA_BLOCK_SIZE> - excecutes loop using the CUDA API
+    
+    CUDA_BLOCK_SIZE - specifies the number of threads in a CUDA thread block
+  */
+  RAJA::forall<RAJA::cuda_exec<CUDA_BLOCK_SIZE>>(
+    RAJA::RangeSegment(0, N), [=] __device__(RAJA::Index_type i) {       
+      C[i] = A[i] + B[i]; 
+    });
+  checkSolution(C, N);
+#endif
+
+  memoryManager::deallocate(A);
+  memoryManager::deallocate(B);
+  memoryManager::deallocate(C);
+
+  return 0;
+}
+
+
+/*
+  RAJA may also be used outside of computational kernels
```

Comment:
Not sure what is meant by this comment. Also, I would add a comment about what this method checks to where it is declared above.  Basically, explain that the correct result for the vector addition is a vector of all zeros (c[i] = a[i]+b[i], where for each i, a[i] = 1, b[i] = -1).  That way folks will know immediately what the example is doing and they won't have to reverse engineer any code to understand it.